nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/preliminary/CpuUsageHealthCheck.java

```diff
@@ -110,50 +92,29 @@ protected HealthCheckResult doCheckInternal() {
             return result;
         }
 
-        final long[] ids = threadBean.getAllThreadIds();
-        for (long id : ids) {
-            if (id == java.lang.Thread.currentThread().getId())
-                continue;
-            final long c = threadBean.getThreadCpuTime(id);
-            final long u = threadBean.getThreadUserTime(id);
-
-            if (c == -1 || u == -1)
-                continue;
-
-            ThreadTimes times = threadTimes.get(id);
-            if (times == null) {
-                times = new ThreadTimes();
-                times.setId(id);
-                times.setStartCpuTime(c);
-                times.setEndCpuTime(c);
-                times.setStartUserTime(u);
-                times.setEndUserTime(u);
-                threadTimes.put(id, times);
-            }
-            else {
-                times.setEndCpuTime(c);
-                times.setEndUserTime(u);
```

Comment:
FYI: Having to duplicate the computation of CPU usage for the statistic provider where such a value was added I needed to understand what is computed. Doing this I discovered that the temporary map for the statistics per thread does not serve any purpose other than totalling the time of all threads which simply could be done by looping and adding their times directly.